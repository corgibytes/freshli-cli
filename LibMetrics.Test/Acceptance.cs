using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using LibMetrics.Languages.Php;
using LibMetrics.Languages.Ruby;
using Xunit;

namespace LibMetrics.Test
{
  [UseReporter(typeof(DiffReporter))]
  public class Acceptance
  {
    [Fact]
    public void RubyGemsWithGitHistory()
    {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();

      var rubyFixturePath = Fixtures.Path("ruby", "nokotest");
      var results = runner.Run(
        rubyFixturePath,
        asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsWithHistoryViaGitHub() {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();

      var repoUrl =
        "https://github.com/corgibytes/libmetrics-fixture-ruby-nokotest";
      var results = runner.Run(repoUrl, asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsFeedbinHistoryViaGitHub()
    {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();

      var repoUrl = "https://github.com/feedbin/feedbin";
      var results = runner.Run(repoUrl, asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void ComposerWithoutGitHistory()
    {
      ManifestFinder.Register<PhpComposerManifestFinder>();

      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "large");
      var results = runner.Run(
        phpFixturePath,
        asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void DrupalComposerWithoutGitHistory()
    {
      ManifestFinder.Register<PhpComposerManifestFinder>();

      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "drupal");
      var results = runner.Run(
        phpFixturePath,
        asOf: new DateTime(2020, 01, 01));

      Approvals.VerifyAll(results, "results");
    }

    }
  }
}
