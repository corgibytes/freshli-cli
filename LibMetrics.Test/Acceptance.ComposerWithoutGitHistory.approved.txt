results[0] = { Date: 1/1/2020 12:00:00 AM, LibYear: { _packagesValues: [ { Name: "aws/aws-sdk-php", Version: "3.130.2", PublishedAt: 12/23/2019 12:00:00 AM, Value: 0 }, { Name: "barryvdh/laravel-dompdf", Version: "v0.8.5", PublishedAt: 8/23/2019 12:00:00 AM, Value: 0 }, { Name: "barryvdh/laravel-snappy", Version: "v0.4.6", PublishedAt: 10/2/2019 12:00:00 AM, Value: 0 }, { Name: "cogpowered/finediff", Version: "0.3.1", PublishedAt: 5/19/2014 12:00:00 AM, Value: 0 }, { Name: "doctrine/cache", Version: "1.10.0", PublishedAt: 11/29/2019 12:00:00 AM, Value: 0 }, { Name: "doctrine/dbal", Version: "v2.10.0", PublishedAt: 11/3/2019 12:00:00 AM, Value: 0 }, { Name: "doctrine/event-manager", Version: "1.1.0", PublishedAt: 11/10/2019 12:00:00 AM, Value: 0 }, { Name: "doctrine/inflector", Version: "1.3.1", PublishedAt: 10/30/2019 12:00:00 AM, Value: 0 }, { Name: "doctrine/lexer", Version: "1.2.0", PublishedAt: 10/30/2019 12:00:00 AM, Value: 0 }, { Name: "dompdf/dompdf", Version: "v0.8.3", PublishedAt: 12/13/2018 12:00:00 AM, Value: 0 }, { Name: "dragonmantank/cron-expression", Version: "v2.3.0", PublishedAt: 3/30/2019 12:00:00 AM, Value: 0 }, { Name: "egulias/email-validator", Version: "1.2.16", PublishedAt: 12/30/2019 12:00:00 AM, Value: 0.0273972602739726 }, { Name: "erusev/parsedown", Version: "1.7.4", PublishedAt: 12/30/2019 12:00:00 AM, Value: 0.7890410958904109 }, { Name: "fideloper/proxy", Version: "4.2.2", PublishedAt: 12/20/2019 12:00:00 AM, Value: 0 }, { Name: "gathercontent/htmldiff", Version: "0.2.1", PublishedAt: 4/15/2015 12:00:00 AM, Value: 0 }, { Name: "guzzlehttp/guzzle", Version: "7.0.0-beta.1", PublishedAt: 12/30/2019 12:00:00 AM, Value: 0.019178082191780823 }, { Name: "guzzlehttp/promises", Version: "v1.3.1", PublishedAt: 12/20/2016 12:00:00 AM, Value: 0 }, { Name: "guzzlehttp/psr7", Version: "1.6.1", PublishedAt: 7/1/2019 12:00:00 AM, Value: 0 }, { Name: "intervention/image", Version: "2.5.1", PublishedAt: 11/2/2019 12:00:00 AM, Value: 0 }, { Name: "knplabs/knp-snappy", Version: "v1.2.0", PublishedAt: 12/2/2019 12:00:00 AM, Value: 0 }, { Name: "laravel/framework", Version: "v6.9.0", PublishedAt: 12/19/2019 12:00:00 AM, Value: 0 }, { Name: "laravel/socialite", Version: "v4.3.1", PublishedAt: 11/26/2019 12:00:00 AM, Value: 0 }, { Name: "league/flysystem", Version: "1.0.62", PublishedAt: 12/29/2019 12:00:00 AM, Value: 0.057534246575342465 }, { Name: "league/flysystem-aws-s3-v3", Version: "1.0.23", PublishedAt: 6/5/2019 12:00:00 AM, Value: 0 }, { Name: "league/oauth1-client", Version: "1.7.0", PublishedAt: 8/16/2016 12:00:00 AM, Value: 0 }, { Name: "monolog/monolog", Version: "2.0.2", PublishedAt: 12/20/2019 12:00:00 AM, Value: 0 }, { Name: "mtdowling/jmespath.php", Version: "2.5.0", PublishedAt: 12/30/2019 12:00:00 AM, Value: 3.073972602739726 }, { Name: "nesbot/carbon", Version: "2.28.0", PublishedAt: 12/16/2019 12:00:00 AM, Value: 0 }, { Name: "onelogin/php-saml", Version: "2.18.1", PublishedAt: 11/25/2019 12:00:00 AM, Value: 0 }, { Name: "opis/closure", Version: "3.5.1", PublishedAt: 11/29/2019 12:00:00 AM, Value: 0 }, { Name: "paragonie/random_compat", Version: "v2.0.18", PublishedAt: 1/3/2019 12:00:00 AM, Value: 0.5068493150684932 }, { Name: "phenx/php-font-lib", Version: "0.5.1", PublishedAt: 9/13/2017 12:00:00 AM, Value: 0 }, { Name: "phenx/php-svg-lib", Version: "v0.3.3", PublishedAt: 9/11/2019 12:00:00 AM, Value: 0 }, { Name: "phpoption/phpoption", Version: "1.7.2", PublishedAt: 12/15/2019 12:00:00 AM, Value: 0 }, { Name: "predis/predis", Version: "v1.1.1", PublishedAt: 6/16/2016 12:00:00 AM, Value: 0 }, { Name: "psr/container", Version: "1.0.0", PublishedAt: 2/14/2017 12:00:00 AM, Value: 0 }, { Name: "psr/http-message", Version: "1.0.1", PublishedAt: 8/6/2016 12:00:00 AM, Value: 0 }, { Name: "psr/log", Version: "1.1.2", PublishedAt: 11/1/2019 12:00:00 AM, Value: 0 }, { Name: "psr/simple-cache", Version: "1.0.1", PublishedAt: 10/22/2017 12:00:00 AM, Value: 0 }, { Name: "ralouphie/getallheaders", Version: "3.0.3", PublishedAt: 3/8/2019 12:00:00 AM, Value: 0 }, { Name: "ramsey/uuid", Version: "3.9.2", PublishedAt: 12/17/2019 12:00:00 AM, Value: 0 }, { Name: "robrichards/xmlseclibs", Version: "1.4.3", PublishedAt: 11/12/2019 12:00:00 AM, Value: 0.019178082191780823 }, { Name: "sabberworm/php-css-parser", Version: "8.3.0", PublishedAt: 2/22/2019 12:00:00 AM, Value: 0 }, { Name: "socialiteproviders/discord", Version: "v2.0.2", PublishedAt: 5/25/2018 12:00:00 AM, Value: 0 }, { Name: "socialiteproviders/gitlab", Version: "v3.1", PublishedAt: 6/27/2018 12:00:00 AM, Value: 0 }, { Name: "socialiteproviders/manager", Version: "v3.4.3", PublishedAt: 9/25/2019 12:00:00 AM, Value: 0 }, { Name: "socialiteproviders/microsoft-azure", Version: "v3.0.0", PublishedAt: 1/25/2017 12:00:00 AM, Value: 0 }, { Name: "socialiteproviders/okta", Version: "v1.1.0", PublishedAt: 9/6/2019 12:00:00 AM, Value: 1.7917808219178082 }, { Name: "socialiteproviders/slack", Version: "v3.1", PublishedAt: 1/11/2019 12:00:00 AM, Value: 0 }, { Name: "socialiteproviders/twitch", Version: "v5.1", PublishedAt: 7/1/2019 12:00:00 AM, Value: 0 }, { Name: "swiftmailer/swiftmailer", Version: "v6.2.3", PublishedAt: 11/12/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/console", Version: "v5.0.2", PublishedAt: 12/17/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/css-selector", Version: "v3.4.37", PublishedAt: 1/1/2020 12:00:00 AM, Value: 0.22465753424657534 }, { Name: "symfony/debug", Version: "v4.4.2", PublishedAt: 12/16/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/error-handler", Version: "v4.4.2", PublishedAt: 12/16/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/event-dispatcher", Version: "v4.4.2", PublishedAt: 11/28/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/event-dispatcher-contracts", Version: "v2.0.1", PublishedAt: 11/18/2019 12:00:00 AM, Value: 0.16986301369863013 }, { Name: "symfony/finder", Version: "v3.4.37", PublishedAt: 1/1/2020 12:00:00 AM, Value: 0.1232876712328767 }, { Name: "symfony/http-foundation", Version: "v4.4.2", PublishedAt: 12/19/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/http-kernel", Version: "v4.4.2", PublishedAt: 12/19/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/mime", Version: "v4.3.10", PublishedAt: 1/1/2020 12:00:00 AM, Value: 0.08767123287671233 }, { Name: "symfony/polyfill-ctype", Version: "v1.13.2", PublishedAt: 11/27/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/polyfill-iconv", Version: "v1.13.2", PublishedAt: 12/3/2019 12:00:00 AM, Value: 0.01643835616438356 }, { Name: "symfony/polyfill-intl-idn", Version: "v1.13.2", PublishedAt: 11/27/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/polyfill-mbstring", Version: "v1.13.0", PublishedAt: 11/27/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/polyfill-php72", Version: "v1.13.2", PublishedAt: 11/27/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/polyfill-php73", Version: "v1.13.2", PublishedAt: 11/27/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/process", Version: "v3.4.37", PublishedAt: 1/1/2020 12:00:00 AM, Value: 0.07123287671232877 }, { Name: "symfony/routing", Version: "v5.0.2", PublishedAt: 12/12/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/service-contracts", Version: "v2.0.1", PublishedAt: 11/18/2019 12:00:00 AM, Value: 0.0958904109589041 }, { Name: "symfony/translation", Version: "v5.0.2", PublishedAt: 12/12/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/translation-contracts", Version: "v2.0.1", PublishedAt: 11/18/2019 12:00:00 AM, Value: 0.16986301369863013 }, { Name: "symfony/var-dumper", Version: "v3.4.37", PublishedAt: 1/1/2020 12:00:00 AM, Value: 0.038356164383561646 }, { Name: "tijsverkoyen/css-to-inline-styles", Version: "2.2.2", PublishedAt: 10/24/2019 12:00:00 AM, Value: 0 }, { Name: "vlucas/phpdotenv", Version: "v4.1.0", PublishedAt: 12/14/2019 12:00:00 AM, Value: 0.2602739726027397 }, { Name: "barryvdh/laravel-debugbar", Version: "v3.2.8", PublishedAt: 8/29/2019 12:00:00 AM, Value: 0 }, { Name: "barryvdh/laravel-ide-helper", Version: "v2.6.6", PublishedAt: 10/30/2019 12:00:00 AM, Value: 0 }, { Name: "barryvdh/reflection-docblock", Version: "v2.0.6", PublishedAt: 12/13/2018 12:00:00 AM, Value: 0 }, { Name: "composer/ca-bundle", Version: "1.2.5", PublishedAt: 12/11/2019 12:00:00 AM, Value: 0 }, { Name: "composer/composer", Version: "1.9.1", PublishedAt: 11/1/2019 12:00:00 AM, Value: 0 }, { Name: "composer/semver", Version: "1.5.0", PublishedAt: 3/19/2019 12:00:00 AM, Value: 0 }, { Name: "composer/spdx-licenses", Version: "1.5.2", PublishedAt: 7/29/2019 12:00:00 AM, Value: 0 }, { Name: "composer/xdebug-handler", Version: "1.4.0", PublishedAt: 11/6/2019 12:00:00 AM, Value: 0 }, { Name: "doctrine/instantiator", Version: "1.3.0", PublishedAt: 10/21/2019 12:00:00 AM, Value: 0 }, { Name: "facade/flare-client-php", Version: "1.3.1", PublishedAt: 12/15/2019 12:00:00 AM, Value: 0 }, { Name: "facade/ignition", Version: "1.13.0", PublishedAt: 11/27/2019 12:00:00 AM, Value: 0 }, { Name: "facade/ignition-contracts", Version: "1.0.0", PublishedAt: 8/30/2019 12:00:00 AM, Value: 0 }, { Name: "filp/whoops", Version: "2.7.0", PublishedAt: 12/29/2019 12:00:00 AM, Value: 0.010958904109589041 }, { Name: "fzaninotto/faker", Version: "v1.9.1", PublishedAt: 12/12/2019 12:00:00 AM, Value: 0 }, { Name: "hamcrest/hamcrest-php", Version: "v2.0.0", PublishedAt: 1/20/2016 12:00:00 AM, Value: 0 }, { Name: "jakub-onderka/php-console-color", Version: "v0.2", PublishedAt: 9/29/2018 12:00:00 AM, Value: 0 }, { Name: "jakub-onderka/php-console-highlighter", Version: "v0.4", PublishedAt: 9/29/2018 12:00:00 AM, Value: 0 }, { Name: "justinrainbow/json-schema", Version: "5.2.9", PublishedAt: 9/25/2019 12:00:00 AM, Value: 0 }, { Name: "laravel/browser-kit-testing", Version: "v5.1.3", PublishedAt: 7/30/2019 12:00:00 AM, Value: 0 }, { Name: "maximebf/debugbar", Version: "v1.16.1", PublishedAt: 11/24/2019 12:00:00 AM, Value: 0.16712328767123288 }, { Name: "mockery/mockery", Version: "1.3.1", PublishedAt: 12/26/2019 12:00:00 AM, Value: 0 }, { Name: "myclabs/deep-copy", Version: "1.9.4", PublishedAt: 12/15/2019 12:00:00 AM, Value: 0 }, { Name: "nunomaduro/collision", Version: "v3.0.1", PublishedAt: 3/7/2019 12:00:00 AM, Value: 0 }, { Name: "phar-io/manifest", Version: "1.0.3", PublishedAt: 7/8/2018 12:00:00 AM, Value: 0 }, { Name: "phar-io/version", Version: "2.0.1", PublishedAt: 7/8/2018 12:00:00 AM, Value: 0 }, { Name: "phpdocumentor/reflection-common", Version: "2.0.0", PublishedAt: 8/7/2018 12:00:00 AM, Value: 0 }, { Name: "phpdocumentor/reflection-docblock", Version: "4.3.4", PublishedAt: 12/28/2019 12:00:00 AM, Value: 0.021917808219178082 }, { Name: "phpdocumentor/type-resolver", Version: "1.0.1", PublishedAt: 8/22/2019 12:00:00 AM, Value: 0 }, { Name: "phploc/phploc", Version: "5.0.0", PublishedAt: 3/16/2019 12:00:00 AM, Value: 0 }, { Name: "phpspec/prophecy", Version: "1.10.1", PublishedAt: 12/22/2019 12:00:00 AM, Value: 0 }, { Name: "phpunit/php-code-coverage", Version: "7.0.9", PublishedAt: 11/20/2019 12:00:00 AM, Value: 0 }, { Name: "phpunit/php-file-iterator", Version: "2.0.2", PublishedAt: 9/13/2018 12:00:00 AM, Value: 0 }, { Name: "phpunit/php-text-template", Version: "1.2.1", PublishedAt: 6/21/2015 12:00:00 AM, Value: 0 }, { Name: "phpunit/php-timer", Version: "2.1.2", PublishedAt: 6/7/2019 12:00:00 AM, Value: 0 }, { Name: "phpunit/php-token-stream", Version: "3.1.1", PublishedAt: 9/17/2019 12:00:00 AM, Value: 0 }, { Name: "phpunit/phpunit", Version: "8.5.1", PublishedAt: 12/25/2019 12:00:00 AM, Value: 0 }, { Name: "scrivo/highlight.php", Version: "v9.17.1.0", PublishedAt: 12/13/2019 12:00:00 AM, Value: 0 }, { Name: "sebastian/code-unit-reverse-lookup", Version: "1.0.1", PublishedAt: 3/4/2017 12:00:00 AM, Value: 0 }, { Name: "sebastian/comparator", Version: "3.0.2", PublishedAt: 7/12/2018 12:00:00 AM, Value: 0 }, { Name: "sebastian/diff", Version: "3.0.2", PublishedAt: 2/4/2019 12:00:00 AM, Value: 0 }, { Name: "sebastian/environment", Version: "4.2.3", PublishedAt: 11/20/2019 12:00:00 AM, Value: 0 }, { Name: "sebastian/exporter", Version: "3.1.2", PublishedAt: 9/14/2019 12:00:00 AM, Value: 0 }, { Name: "sebastian/finder-facade", Version: "1.2.2", PublishedAt: 11/18/2017 12:00:00 AM, Value: 0 }, { Name: "sebastian/global-state", Version: "3.0.0", PublishedAt: 2/1/2019 12:00:00 AM, Value: 0 }, { Name: "sebastian/object-enumerator", Version: "3.0.3", PublishedAt: 8/3/2017 12:00:00 AM, Value: 0 }, { Name: "sebastian/object-reflector", Version: "1.1.1", PublishedAt: 3/29/2017 12:00:00 AM, Value: 0 }, { Name: "sebastian/recursion-context", Version: "3.0.0", PublishedAt: 3/3/2017 12:00:00 AM, Value: 0 }, { Name: "sebastian/resource-operations", Version: "2.0.1", PublishedAt: 10/4/2018 12:00:00 AM, Value: 0 }, { Name: "sebastian/type", Version: "1.1.3", PublishedAt: 7/2/2019 12:00:00 AM, Value: 0 }, { Name: "sebastian/version", Version: "2.0.1", PublishedAt: 10/3/2016 12:00:00 AM, Value: 0 }, { Name: "seld/jsonlint", Version: "1.7.2", PublishedAt: 10/24/2019 12:00:00 AM, Value: 0 }, { Name: "seld/phar-utils", Version: "1.0.1", PublishedAt: 10/13/2015 12:00:00 AM, Value: 0 }, { Name: "squizlabs/php_codesniffer", Version: "3.5.3", PublishedAt: 12/3/2019 12:00:00 AM, Value: 0 }, { Name: "symfony/dom-crawler", Version: "v3.4.37", PublishedAt: 1/1/2020 12:00:00 AM, Value: 0.17534246575342466 }, { Name: "symfony/filesystem", Version: "v4.4.1", PublishedAt: 11/26/2019 12:00:00 AM, Value: 0 }, { Name: "theseer/fdomdocument", Version: "1.6.6", PublishedAt: 6/30/2017 12:00:00 AM, Value: 0 }, { Name: "theseer/tokenizer", Version: "1.1.3", PublishedAt: 6/13/2019 12:00:00 AM, Value: 0 }, { Name: "webmozart/assert", Version: "1.6.0", PublishedAt: 11/24/2019 12:00:00 AM, Value: 0 }, { Name: "wnx/laravel-stats", Version: "v2.0.0", PublishedAt: 9/1/2019 12:00:00 AM, Value: 0 },  ], Total: 7.917808219178081 } }
